### PR TITLE
Updated Cosmos DB Partitioned following feedback

### PR DESCRIPTION
**Pull request by garypretty**
- Amends container creation to disable indexing
- Switch to using ReadItemAsync instead of Query
- Switch PartitionKey to use Id instead of RealId

### CODE REVIEW

**Comment by garypretty** [comment]
Thanks @christopheranderson - I couldn't see a way of specifying the throughput when defining my container.  I don't believe this is an issue anyway as the user can manage afterwards in Azure if needed and I will ensure it is added to the sample comments as well. Cheers again!

File: libraries/Microsoft.Bot.Builder.Azure/CosmosDbPartitionedStorage.cs [hunk 1]
```diff
@@ -216,13 +231,14 @@ private async Task InitializeAsync()
...
-                    _container = await _client
+                    var containerResponse = await _client
                         .GetDatabase(_cosmosDbStorageOptions.DatabaseId)
-                        .CreateContainerIfNotExistsAsync(
-                            _cosmosDbStorageOptions.ContainerId,
-                            DocumentStoreItem.PartitionKeyPath,
-                            _cosmosDbStorageOptions.ContainerThroughput)
+                        .DefineContainer(_cosmosDbStorageOptions.ContainerId, DocumentStoreItem.PartitionKeyPath)
+                        .WithIndexingPolicy().WithAutomaticIndexing(false).WithIndexingMode(IndexingMode.None).Attach()
+                        .CreateIfNotExistsAsync()
```
Comment (j82w) [inline libraries/Microsoft.Bot.Builder.Azure/CosmosDbPartitionedStorage.cs]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.database.definecontainer?view=azure-dotnet#Microsoft_Azure_Cosmos_Database_DefineContainer_System_String_System_String_

```suggestion
                        .CreateIfNotExistsAsync(_cosmosDbStorageOptions.ContainerThroughput)
```

Comment (garypretty) [reply libraries/Microsoft.Bot.Builder.Azure/CosmosDbPartitionedStorage.cs]: @j82w Ah! Thank you! I will update.

**Comment by mdrichardson** [comment]
@gary Should the tests include [not throwing when deleting non-existing item](https://github.com/microsoft/botbuilder-dotnet/blob/master/tests/Microsoft.Bot.Builder.Tests/StorageBaseTests.cs#L204)? I'm not sure if it throws a 404 on the dotnet side, but I know it does in Node.

**Comment by garypretty** [comment]
@mdrichardson ill check. Good shout I think. I'll add a test. We have a read unknown so we should probably have a delete unknown test too.

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.5.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.5.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.5.3)

**Pull request merged**